### DISCUSSION TITLE
Hybrid mode:- Data plane cannot communicate with the control plane in docker compose

### DISCUSSION THREAD

**Discussion by suvsap** [Kong/kong#10244] [Help] [answered]
Hi,

I have two nodes as part of docker swarm

Node 1:- hosts the Postgress database installed in a VM
Node 2:- hosts the Controlplane, Dataplane and Kong-migration to bootstrap the postgress database with controlplane

Control-plane parameters:-

      KONG_ROLE: control_plane
      KONG_CLUSTER_CERT: /cert/cluster.crt
      KONG_CLUSTER_CERT_KEY: /cert/cluster.key
      KONG_ADMIN_LISTEN: 0.0.0.0:8001, 0.0.0.0:8444 ssl
      KONG_ADMIN_GUI_LISTEN: 0.0.0.0:8445 ssl
      KONG_DATABASE: postgres
      KONG_PG_HOST: 10.143.4.62
      KONG_PG_USER: kong
      KONG_PROXY_LISTEN: "off"
      KONG_STREAM_LISTEN: "off"
      KONG_PG_PASSWORD_FILE: /run/secrets/kongtestpwd
      KONG_PG_DATABASE: kong
      #KONG_SSL: “on”
      #KONG_SSL_CERT: /cert/test/testcertificate.crt
      #KONG_SSL_CERT_KEY: /cert/test/testcertificate.key
      KONG_ADMIN_SSL_CERT: /cert/test1/test1certificate
      KONG_ADMIN_SSL_CERT_KEY: /cert/test1/test1certificate.key
      KONG_ADMIN_GUI_SSL_CERT: /cert/test1/tes1certificate.crt
      KONG_ADMIN_GUI_SSL_CERT_KEY: /cert/test1/testcertificate.key
      KONG_ADMIN_API_URI: https://test-api.com:8444
      KONG_ADMIN_GUI_URL:  https://test.com:8445
      KONG_LICENSE_PATH: /kongenterprise/license.json
      KONG_CLUSTER_LISTEN: 0.0.0.0:8005 ssl
      KONG_CLUSTER_TELEMETRY_LISTEN: 0.0.0.0:8006 ssl
      
      DataPlane Parameters:-
      
      
            KONG_ROLE: data_plane
      KONG_CLUSTER_CONTROL_PLANE: 127.0.0.1:8005
      KONG_CLUSTER_TELEMETRY_ENDPOINT: 127.0.0.1:8006
      KONG_KONG_TRUSTED_IPS: 0.0.0.0/0,::0
      KONG_CLUSTER_LISTEN: 0.0.0.0:8005 ssl
      KONG_CLUSTER_TELEMETRY_LISTEN: 0.0.0.0:8006 ssl
      KONG_DATABASE: "off"
      KONG_ADMIN_LISTEN: "off"
      KONG_PROXY_LISTEN: 0.0.0.0:8000, 0.0.0.0:8443 ssl
      KONG_SSL: “on”
      KONG_SSL_CERT: /cert/test1/test1certificate.crt
      KONG_SSL_CERT_KEY: /cert/test1/test1certificate.key
      # ssl_cipher_suite: "intermediate"
      KONG_CLUSTER_CERT: /cert/cluster.crt
      KONG_CLUSTER_CERT_KEY: /cert/cluster.key
      KONG_LUA_SSL_TRUSTED_CERTIFICATE: /cert/cluster.crt
      
      When i deploy the docker stack they are running successfully, but the data plane gives the below error:-
      
      2023/02/06 15:23:26 [error] 2139#0: *1988 [lua] init.lua:370: connection to control plane wss://127.0.0.1:8006/v1/ingest?node_id=57c1abd6-8685-4d9c-b53d-bd47563627f3&node_hostname=2a327874a31f&node_version=3.0.2.0 broken: failed to connect: connection refused retrying after 9 seconds, context: ngx.timer
2023/02/06 15:23:32 [error] 2139#0: *1997 [lua] data_plane.lua:111: communicate(): [clustering] connection to control plane wss://127.0.0.1:8005/v1/outlet?node_id=57c1abd6-8685-4d9c-b53d-bd47563627f3&node_hostname=2a327874a31f&node_version=3.0.2.0 broken: failed to connect: connection refused (retrying after 9 seconds) [127.0.0.1:8005], context: ngx.timer

Kindly help

Thanks

**Comment by hanshuebner** [comment, marked as answer]
Hello @suvsap,

sorry for the late reply.  If I understand correctly, you're running your data and control planes in separate docker containers.  When starting the data plane, however, you are supplying `localhost` as the host running the data plane:
`KONG_CLUSTER_CONTROL_PLANE: 127.0.0.1:8005`
You will need to use the hostname of the control plane in the virtual docker network, however.

Maybe I'm misunderstanding the configuration, though.

Please let us know if the problem persists.
-Hans
